### ISSUE TITLE
Suggestions for panning and zooming experience

### ISSUE THREAD

**Issue by CodingOctocat** [TBertuzzi/Bertuzzi.MAUI.PinchZoomImage#7]
If you use this control as an image viewer, it's a bit of a less than stellar experience.

**Boundary processing during panning**

Assuming blue is the PinchZoom container and orange is the image:

1. When the width of the image is less than or equal to the width of the container, it should not be able to be panned left or right (the same applies when the height is less than the height of the container)
2. When the width of the image is greater than the width of the container, it can be panned left and right, but was to the container boundary (the same when the height is greater than the height of the container)
3. When the width and height of the image are larger than the container, the boundary situation is shown in Figure

---
**Scaling issues:**

1. When the width "or/and" height of the image is less than the width "or/and" height of the container, the scaling should keep the image centered.
2. When the image width "or/and" height is greater than the container width "or/and" height, it should be scaled at the pinch (to make it intuitive for the user), but should not violate the panning boundary rule above.

To summarize: we might as well switch perspectives and find that when the image size is larger than the container, it is equivalent to the container (left and right "or/and" top and bottom edges) being confined to the image

![image](https://github.com/TBertuzzi/Bertuzzi.MAUI.PinchZoomImage/assets/7220248/f5b7652f-502b-4dbb-a2a3-a22424763e14)

**Comment by CodingOctocat** [comment]
Hi~ o(*￣▽￣*)ブ, I implemented a control with better user experience: [MauiPanPinchContainer](https://github.com/CodingOctocat/MauiPanPinchContainer), hope this can help more people!

**Comment by bhavanesh2001** [comment]
https://github.com/bhavanesh2001/Plugin.Maui.ZoomView
